### PR TITLE
Problem: recent ethermint and cosmos-sdk fixes not included

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #288
Solution:
- update ethermint to v0.10.0 with ibc-go reverted to v2.0.2.
- update cosmos-sdk to v0.45.0
- ibc-go: v2.0.2-hooks
- gravity-bridge module: v0.3.4
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod
Comment (yihuang) [inline go.mod:158]: TODO, update to crypto-org-chain later.

File: go.mod
Comment (yihuang) [inline go.mod:168]: TODO, update after merged, https://github.com/PeggyJV/gravity-bridge/pull/322

File: go.mod
Comment (yihuang) [inline go.mod:166]: TODO, backported https://github.com/cosmos/cosmos-sdk/pull/10814 to 0.44.x, update after merged.

File: go.mod
Comment (yihuang) [inline go.mod:164]: ethermint main + https://github.com/tharsis/ethermint/pull/869 + statedb refactoring.

File: go.mod
Comment (yihuang) [inline go.mod:164]: ethermint main + https://github.com/tharsis/ethermint/pull/869 + https://github.com/tharsis/ethermint/pull/871 + statedb refactoring.
TODO move to crypto-org-chain and make a tag after finalized.

File: go.mod
Comment (thomas-nguy) [reply go.mod:158]: I have created a tag in crypto-org-chain for you to use it

File: integration_tests/test_basic.py
Comment (yihuang) [inline integration_tests/test_basic.py:111]: the error is not an exception anymore, but a zero receipt status now. I guess it's because error message changed.

File: go.mod
Comment (yihuang) [inline go.mod:162]: TODO: wait for merge: https://github.com/cosmos/cosmos-sdk/pull/10814

**Comment by lhypj** [comment]
when finished.

**Comment by JayT106** [comment]
LGTM

File: app/app.go
Comment (yihuang) [inline app/app.go:517]: cosmos-sdk 0.45 require we pass full module list to the apis: `SetOrderBeginBlockers`/`SetOrderEndBlockers`/`SetInitGenesis`, so it's good to share the module list.

File: go.mod
Comment (yihuang) [inline go.mod:159]: this dependency is gone now even without this replacement.

File: go.mod
Comment (yihuang) [inline go.mod:154]: it's fine without replacement now.

**Review by calvinaco — APPROVED** [review]
LGTM

**Pull request merged**
